File: backend/app/models/service.py
from uuid import UUID
from typing import Dict, Any
from pydantic import BaseModel
from datetime import datetime


class BaseService(BaseModel):
    name: str
    description: str


class ServiceInDB(BaseService):
    id: UUID
    created_at: datetime


class ServiceCreate(BaseService):
    class Config:
        json_schema_extra: Dict[str, Any] = {
            "example": {
                "name": "plumbing",
                "description": "",
            }
        }


class ServiceUpdate(BaseModel):
    description: str

    class Config:
        json_schema_extra: Dict[str, Any] = {
            "example": {
                "description": "",
            }
        }


class ServiceResponse(ServiceInDB):
    class Config:
        json_schema_extra: Dict[str, Any] = {
            "example": {
                "id": "123e4567-e89b-12d3-a456-426655440000",
                "name": "plumbing",
                "description": "",
                "created_at": "2025-06-01T00:00:00",
            }
        }

File: frontend/src/lib/types/service.ts
import { z } from "zod";

const BaseServiceSchema = z.object({
    name: z.string(),
    description: z.string(),
})

export const ServiceCreateSchema = BaseServiceSchema.extend({})

export const ServiceUpdateSchema = z.object({
    name: z.string().optional(),
    description: z.string().optional(),
})

export const ServiceResponseSchema = BaseServiceSchema.extend({
    id: z.string().uuid(),
    created_at: z.string().datetime(),
})

export type ServiceCreate = z.infer<typeof ServiceCreateSchema>;
export type ServiceUpdate = z.infer<typeof ServiceUpdateSchema>;
export type ServiceResponse = z.infer<typeof ServiceResponseSchema>;
